dask_cuml/linear_model/linear_regression.py

```diff
@@ -0,0 +1,560 @@
+                                     unique=np.random.randint(0, 1e6),
+                                     workers=[a_worker])
+                       for a_worker, future in not_on_worker]
+
+        raw_arrays = [future for a_worker, future in on_worker]
+
+        # IPC Handles are loaded in separate threads on worker so they can be
+        # used to make calls through cython
+        # Calls _predict_on_worker defined in the bottom
+        ret = client.submit(_predict_on_worker, (ipc_handles, raw_arrays),
+                            self.intercept, self._build_params_map(),
+                            workers=[exec_node])
+
+        yield wait(ret)
+
+        dfs = [client.submit(series_on_worker, f, worker, loc_dict,
+                             X_df.npartitions, X_df, workers=[worker])
+               for worker, f in input_devarrays]
+
+        return dfs
+
+    def _build_host_dict(self, gpu_futures, client):
+
+        who_has = client.who_has(gpu_futures)
+
+        key_to_host_dict = {}
+        for key in who_has:
+            key_to_host_dict[key] = parse_host_port(who_has[key][0])
+
+        hosts_to_key_dict = {}
+        for key, host in key_to_host_dict.items():
+            if host not in hosts_to_key_dict:
+                hosts_to_key_dict[host] = set([key])
+            else:
+                hosts_to_key_dict[host].add(key)
+
+        workers = [key[0] for key in list(who_has.values())]
+        return build_host_dict(workers)
+
+
+def _fit_on_worker(data, params):
+    ipc_dev_list, devarrs_dev_list = data
+
+    # Open 1 ipc thread per device
+    open_ipcs = []
+    for p, dev in ipc_dev_list:
+        arrs = []
+        for x, y, coef in p:
+            arrs.append(x)
+            arrs.append(y)
+            arrs.append(coef)
+        ipct = new_ipc_thread(arrs, dev)
+        open_ipcs.append(ipct)
+
+    # alloc_info = list(itertools.chain([t.info() for t in open_ipcs]))
+    # alloc_info.extend(
+    #     list(itertools.chain(
+    #         [[(build_alloc_info(X)[0], build_alloc_info(y)[0],
+    #            build_alloc_info(coef)[0]) for X, y, coef in p]
+    #          for p, dev in devarrs_dev_list])))
+
+    alloc_info = []
+    for t in open_ipcs:
+        outsiders = t.info()
+        triplet = []
+        for i in range(0, len(outsiders), 3):
+            triplet.append(outsiders[i])
+            triplet.append(outsiders[i+1])
+            triplet.append(outsiders[i+2])
+            alloc_info.append(triplet)
+
+    for p, dev in devarrs_dev_list:
+        locals = []
+        for X, coef, pred in p:
+            locals.append(build_alloc_info(X)[0])
+            locals.append(build_alloc_info(coef)[0])
+            locals.append(build_alloc_info(pred)[0])
+        alloc_info.append(locals)
+
+    try:
+        from cuml.linear_model.linear_regression_mg import LinearRegressionMG as cuOLS
+        ols = cuOLS()
+        intercept = ols._fit_dask(alloc_info, params)
```

Comment:
Maybe rename this `fit_mg()` on the cuml side?

Reply:
Same as above